applications/HDF5Application/custom_io/hdf5_file.h

```diff
@@ -85,216 +99,409 @@ class KRATOS_API(HDF5_APPLICATION) File
     File& operator=(File&& rOther);
 
     ///@}
-    ///@name Operations
+    ///@name Public operations
     ///@{
 
-    /// Check if path exists in HDF5 file.
+    /**
+     * @brief Checks if the given @ref rPath exists.
+     *
+     * @throws If a string with invalid characters are given.
+     *
+     * @param rPath     Path to be checked for.
+     * @return true     If @ref rPath exists.
+     * @return false    If @ref rPath does not exist.
```

Comment:
There should only be a single `return` tag. This also applies to the rest of the docstrings.

```suggestion
     * @return true     True if @a rPath exists, otherwise false.
```

Reply:
done :)